src/Groups/sub.jl

```diff
@@ -26,25 +26,16 @@ export
 ################################################################################
 
 function _as_subgroup_bare(G::T, H::GapObj) where T
-  if T==PermGroup
-    H1 = T(H, G.deg)
-  elseif T<:MatrixGroup
-    H1 = MatrixGroup(G.deg,G.ring)
-    H1.ring_iso = G.ring_iso
-    H1.X = H
-  else
-    H1 = T(H)
-  end
-  return H1
+  return T(H)
 end
```

Comment:
I understand the proposed change in such a way that individual methods for `_as_subgroup_bare` are better than a chain of `if` statements.
In this case, I am suspicious whether this is a good idea: The `else` case looks problematic; if already permutation groups are non-generic then new types of groups are also likely to require more information than just the type of `G`.

Reply:
We discussed this on gather. Basically, these internal helper functions need to be documented to explain what's going on, and probably a guide on how to implement more group types would be good.

I'll merge this PR as-is for now and submit the changes we discussed in a separate PR once they are ready